cristin-project/src/main/java/no/unit/nva/cristin/projects/common/QueryProject.java

```diff
@@ -0,0 +1,90 @@
+import static no.unit.nva.cristin.model.Constants.PROJECTS_PATH;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.BIOBANK;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.IGNORE_PATH_PARAMETER_INDEX;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.KEYWORD;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.LANGUAGE;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.ORGANIZATION;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.PARTICIPANT;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.PATH_PROJECT;
+import static no.unit.nva.cristin.projects.common.ParameterKeyProject.STATUS;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import no.unit.nva.cristin.model.CristinQuery;
+import no.unit.nva.cristin.projects.model.nva.ProjectStatus;
+import nva.commons.core.paths.UriWrapper;
+
+public class QueryProject extends CristinQuery<ParameterKeyProject> {
+    // https://api.cristin-test.uio.no/v2/doc/index.html#GETprojects
+
+    public static QueryBuilderProject builder() {
+        return new QueryBuilderProject();
+    }
+
+    /**
+     * Creates a URI to Cristin project with specific ID and language.
+     *
+     * @param id       Project ID to lookup in Cristin
+     * @param language what language we want some of the result fields to be in
+     * @return an URI to Cristin Projects with ID and language parameters
+     */
+    public static URI fromIdAndLanguage(String id, String language) {
+        return
+            UriWrapper.fromUri(CRISTIN_API_URL)
+                .addChild(PROJECTS_PATH)
+                .addChild(id)
+                .addQueryParameters(Map.of(LANGUAGE.getKey(), language))
+                .getUri();
+    }
+
+    @Override
+    protected String toCristinQueryValue(Entry<ParameterKeyProject, String> entry) {
+        if (entry.getKey().equals(BIOBANK) || entry.getKey().equals(KEYWORD) || entry.getKey().equals(PARTICIPANT)) {
+            final var key = entry.getKey().getKey() + EQUAL_OPERATOR;
+            return Arrays.stream(entry.getValue().split(","))
+                       .collect(Collectors.joining("&" + key));
+        }
+        var value = entry.getKey().isEncode()
+                        ? encodeUTF(entry.getValue())
+                        : entry.getValue();
+
+        if (entry.getKey().equals(STATUS)) {
+            return ProjectStatus.valueOf(value).getCristinStatus();
+        }
+        return entry.getKey().equals(ORGANIZATION) && entry.getValue().matches(PATTERN_IS_URL)
+                   ? getUnitIdFromOrganization(value)
+                   : value;
+    }
+
+    @Override
+    protected String getNvaPathItem(int pathSize, Entry<ParameterKeyProject, String> entry) {
+        var isProjects = entry.getKey().equals(PATH_PROJECT) && pathSize > 1;
```

Comment:
what does pathSize do?

Reply:
I suggest that we move this to a refactoring, as it involves quite a few classes.
(splitting project and organization into two separate queries)
That will also reduce other code smells as well.